src/api/utils_test.go

```diff
@@ -75,3 +75,45 @@ func TestHandlesThrownError(t *testing.T) {
 		t.Errorf("Expected error message %q != %q", ServerError.Error(), actual_message)
 	}
 }
+
+func TestGetsProxiedAddr(t *testing.T) {
+	r, _ := http.NewRequest("GET", "/", nil)
+	r.RemoteAddr = "1.1.1.1:1234"
+	srv := &Server{proxy_addr: ""}
+
+	// Non-proxied request should work
+	addr, err := getProxiedClientAddr(srv, r)
+	if err != nil {
+		t.Error(err)
+	} else if addr != "1.1.1.1" {
+		t.Errorf("Wrong proxy address: %q", addr)
+	}
+
+	// Simulate a proxied request
+	r.Header.Set("X_HTTP_REAL_IP", "2.2.2.2")
+
+	// Works without a proxy address set
```

Comment:
the header should be ignored or we return an error when someone attempt to set the client address through the header without using the proxy.